vms/evm/subscriber.go

```diff
@@ -117,6 +125,29 @@ func (s *subscriber) processBlockRange(
 	return nil
 }
 
+func (s *subscriber) getHeaderByNumberRetryable(headerNumber *big.Int) (*types.Header, error) {
+	var err error
+	var header *types.Header
+	attempt := 1
+	for {
```

Comment:
why don't we do a regular for attempt=1; attempt < retryAttempts; attempt ++ here

Reply:
matching the way that subscriber reattempts right below this